server/src/v2/routes/secret/submit.js

```diff
@@ -41,9 +60,12 @@ export async function secretSubmit(req, res, next) {
     if (pwned && !allowInsecurePassphrase)
       return next({message: 'Passphrase has been pwned (leaked online); please use something else.'})
 
-    if (hasProperty(req.body, 'method') && !METHODS.includes(req.body.method))
-      return next({message: 'Param `method` must be one of: ' + METHODS.join(', ')})
-    var method = hasProperty(req.body, 'method') ? req.body.method.toLowerCase() : DEFAULT_METHOD
+    // Check if the Method being used requires a passphrase
+    if ( requiresPassphrase ) {
```

Comment:
```suggestion
    if (method !== 'publickey') {
```

This variable seems unneccessary when publickey is the only instance where this would ever be needed

Reply:
I was putting this in place just to ensure future proofing.

Reply:
I would rewrite lines 27 to 68 as the following
```js
var secretId = cipher.generateIdentifier()

if (!hasProperty(req.body, 'text') && !(req.files && Object.keys(req.files).length))
  return next({message: 'Missing required body param: `text` OR `file` (must use one).'})

if (hasProperty(req.body, 'method') && !METHODS.includes(req.body.method))
  return next({message: 'Param `method` must be one of: ' + METHODS.join(', ')})
var method = hasProperty(req.body, 'method') ? req.body.method.toLowerCase() : DEFAULT_METHOD
var usingPassphraselessMethod = METHODS_NO_PASSPHRASE.includes(method)

// passphrase referes to both the password and public key if the public key item is selected
if (!hasProperty(req.body, 'passphrase'))
  return next({message: 'Missing required body param: `passphrase`.'})
var passphrase = req.body.passphrase.toString()

var pubkeyRegex = /-{5}BEGIN PUBLIC KEY-{5}.*-{5}END PUBLIC KEY-{5}/s
if (method === 'publickey' && !pubkeyRegex.test(passphrase)) {
  return next({message: 'Public key format does not match the format of X.509 SubjectPublicKeyInfo/OpenSSL PEM public key.'})
}

if (hasProperty(req.body, 'allow_insecure_passphrase') && !isBooleanProperty(req.body.allow_insecure_passphrase))
  return next({message: 'Param `allow_insecure_passphrase` must be of type Boolean.'})
var enforcedSecurePassphrase = hasProperty(req.body, 'allow_insecure_passphrase') ? parseInsecureBoolean(req.body.allow_insecure_passphrase) : true

if (enforcedSecurePassphrase && !usingPassphraselessMethod) {
  var pwned = await pwnedPassphrase(passphrase)
  if (pwned && !allowInsecurePassphrase)
    return next({message: 'Passphrase has been pwned (leaked online); please use something else.'})
}
```

Reply:
eb9bf13